addons/mail/static/src/components/thread_viewer/thread_viewer_tests.js

```diff
@@ -476,6 +476,33 @@ QUnit.test('show message subject if thread is mailing channel', async function (
     )
 });
 
+QUnit.test('should not display user notification messages in non-mailboxes', async function (assert) {
+    assert.expect(1);
+
+    this.data['mail.message'].records = [{
+        author_id: [7, "Demo"],
+        body: "<p>User notification</p>",
+        date: "2019-04-20 11:00:00",
+        id: 102,
+        message_type: 'user_notification',
+        model: 'res.partner',
+        res_id: 100,
+    }];
+    await this.start();
+    const thread = this.env.models['mail.thread'].create({
+        id: 100,
+        model: 'res.partner',
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({ thread: [['link', thread]] });
+    await this.createThreadViewerComponent(threadViewer);
+
+    assert.containsNone(
+        document.body,
+        '.o_Message',
+        "should display no messages"
+    );
+});
+
```

Comment:
This test is too weak because removing the message from `this.data` would lead to a success as well, or having an issue in `message_fetch` mock, or plenty of other causes.

There must be at least one assertion to ensure the setup is correct, otherwise we might as well not test the feature at all since the test is likely going to be broken unnoticed at some point.

Reply:
I'm good with test right now, sure there could be issue with mock server or data, but I don't think the test must adapt from these potential weak parts, and instead we must be meticulous in writing correct tests.